PluginCore/PluginCore/Helpers/ShortcutKeysConverter.cs

```diff
@@ -0,0 +1,224 @@
+            Add(Keys.NumPad5, "Num 5");
+            Add(Keys.NumPad6, "Num 6");
+            Add(Keys.NumPad7, "Num 7");
+            Add(Keys.NumPad8, "Num 8");
+            Add(Keys.NumPad9, "Num 9");
+            Add(Keys.Multiply, "Num *");
+            Add(Keys.Add, "Num +");
+            Add(Keys.Subtract, "Num -");
+            Add(Keys.Decimal, "Num .");
+            Add(Keys.Divide, "Num /");
+            Add(Keys.OemSemicolon, ";"); // Keys.Oem1
+            Add(Keys.Oemplus, "=");
+            Add(Keys.Oemcomma, ",");
+            Add(Keys.OemMinus, "-");
+            Add(Keys.OemPeriod, ".");
+            Add(Keys.OemQuestion, "/"); // Keys.Oem2
+            Add(Keys.Oemtilde, "~"); // Keys.Oem3
+            Add(Keys.OemOpenBrackets, "["); // Keys.Oem4
+            Add(Keys.OemPipe, "\\"); // Keys.Oem5
+            Add(Keys.OemCloseBrackets, "]"); // Keys.Oem6
+            Add(Keys.OemQuotes, "\""); // Keys.Oem7
+            Add(Keys.OemBackslash, "\\"); // Keys.Oem102
+            Add(Keys.Shift, "Shift");
+            Add(Keys.Control, "Ctrl");
+            Add(Keys.Alt, "Alt");
+        }
+
+        private static void Add(Keys key, string name)
+        {
+            names[key] = name;
+            keys[name] = key;
+        }
+
+        private static string ConvertToStringInternal(Keys keys)
+        {
+            // For performance reasons, instead of using a string or StringBuilder buffer and appending
+            // text such as Ctrl, Alt and Shift on it, use the string concatenation to utilize the compiler optimization,
+            // which turns them into string.Concat() calls.
+            if ((keys & Keys.Control) == Keys.Control)
+            {
+                if ((keys & Keys.Alt) == Keys.Alt)
+                {
+                    if ((keys & Keys.Shift) == Keys.Shift)
+                    {
+                        return Ctrl + Alt + Shift + GetString(keys & Keys.KeyCode);
+                    }
+                    return Ctrl + Alt + GetString(keys & Keys.KeyCode);
+                }
+                if ((keys & Keys.Shift) == Keys.Shift)
+                {
+                    return Ctrl + Shift + GetString(keys & Keys.KeyCode);
+                }
+                return Ctrl + GetString(keys & Keys.KeyCode);
+            }
+            if ((keys & Keys.Alt) == Keys.Alt)
+            {
+                if ((keys & Keys.Shift) == Keys.Shift)
+                {
+                    return Alt + Shift + GetString(keys & Keys.KeyCode);
+                }
+                return Alt + GetString(keys & Keys.KeyCode);
+            }
+            if ((keys & Keys.Shift) == Keys.Shift)
+            {
+                return Shift + GetString(keys & Keys.KeyCode);
+            }
+            return GetString(keys);
+        }
+
+        private static unsafe ShortcutKeys ConvertFromStringInternal(string value)
+        {
+            // For performance reasons, use an unsafe context and char pointers.
```

Comment:
Can you quantify the benefit of using an unsafe pointer?
Is that something we should consider for the AS/Haxe model parser?